### PR TITLE
Tmz 3.0.0 gcd lcm - closes #1203

### LINKED ISSUE #1203: feature: GGT, KGV

### PR DESCRIPTION
**Pull request by t-m-z**
greatest common divisor - ggT
least commen multiplier - kgV

### CODE REVIEW

File: lib/tools/science_and_technology/gcd_lcm/logic/gcd_lcm.dart [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import 'dart:core';
+
+int gcd(int a, int b) {
```
Comment (S-Man42) [inline lib/tools/science_and_technology/gcd_lcm/logic/gcd_lcm.dart:3]: For me functions for the common math lib.

File: test/tools/science_and_technology/gcd_lcm/logic/gcd_lcm_test.dart [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import "package:flutter_test/flutter_test.dart";
+import 'package:gc_wizard/tools/science_and_technology/gcd_lcm/logic/gcd_lcm.dart';
+
+void main() {
+  group("gcd", () {
+    List<Map<String, Object?>> _inputsToExpected = [
+      {'input_a' : 0, 'input_b' : 0, 'expectedOutput' : 0},
```
Comment (S-Man42) [inline test/tools/science_and_technology/gcd_lcm/logic/gcd_lcm_test.dart:7]: What about negative values?

File: lib/tools/science_and_technology/gcd_lcm/widget/gcd_lcm.dart [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import 'package:flutter/material.dart';
+import 'package:gc_wizard/application/i18n/app_localizations.dart';
+import 'package:gc_wizard/common_widgets/outputs/gcw_columned_multiline_output.dart';
+import 'package:gc_wizard/common_widgets/outputs/gcw_default_output.dart';
+import 'package:gc_wizard/common_widgets/spinners/gcw_integer_spinner.dart';
+import 'package:gc_wizard/tools/science_and_technology/gcd_lcm/logic/GCD_LCM.dart';
+
+
+class GCDLCM extends StatefulWidget {
```
Comment (S-Man42) [inline lib/tools/science_and_technology/gcd_lcm/widget/gcd_lcm.dart:9]: I'd prefer two different tools for two different functions

**Comment by t-m-z** [comment]
reafctor done

File: assets/searchstrings/common.json [hunk 4]
```diff
@@ -175,6 +175,7 @@
   "games_scrabble": "scrabble",
   "games_sudokusolver": "sudoku",
   "gccode": "geocachingcode geocachecode gccode base31",
+  "gcd_lcm": "greatestcommondivision gcd leastvommonmultiple lcm",
```
Comment (S-Man42) [inline assets/searchstrings/common.json:178]: This is not for common, this has to be translated

**Comment by S-Man42** [comment]
grafik
Why are you using a list output widget, not a normal one? You don't have a list, don't you?

**Comment by S-Man42** [comment]
grafik
For some reason, not clear to me at all, i can't find anything when searching for lcm

**Comment by S-Man42** [comment]
Currently you are returning 0 when an input number is 0. But in my opinion, thats wrong, isn't it. 0 can't be a divider any time because that would give a division by zero. So in my opinion, to be correct. That's true for multiple. 0 can't be a multiple of anything. Because that would mean, this number could be divided by 0 as well. So 0 is no valid input for LCM or GCD. Maybe here a message would be more correct, what do you think?

File: assets/searchstrings/en.json [hunk 5]
```diff
@@ -240,6 +241,7 @@
   "languagegames_piglatin": "piglatin",
   "languagegames_robberlanguage": "robberlanguage",
   "languagegames_spoonlanguage": "spoonlanguage spoonspeak",
+  "lcd": "least common multiplier",
```
Comment (S-Man42) [inline assets/searchstrings/en.json:244]: lcd? isn't it lcm? and here the searchstring lcm is missing

**Comment by t-m-z** [comment]
> Currently you are returning 0 when an input number is 0. But in my opinion, thats wrong, isn't it. 0 can't be a divider any time because that would give a division by zero. So in my opinion, to be correct. That's true for multiple. 0 can't be a multiple of anything. Because that would mean, this number could be divided by 0 as well. So 0 is no valid input for LCM or GCD. Maybe here a message would be more correct, what do you think?

according to Wikipedia: 
for the case m=0 or n=0 is the lcm definded as lcm(m; n) = 0
Harald Scheid: Einführung in die Zahlentheorie. Klett Verlag, Stuttgart, 1972, [ISBN 3-12-983240-8](https://de.wikipedia.org/wiki/Spezial:ISBN-Suche/3129832408), S. 79.

**Comment by t-m-z** [comment]
> Currently you are returning 0 when an input number is 0. But in my opinion, thats wrong, isn't it. 0 can't be a divider any time because that would give a division by zero. So in my opinion, to be correct. That's true for multiple. 0 can't be a multiple of anything. Because that would mean, this number could be divided by 0 as well. So 0 is no valid input for LCM or GCD. Maybe here a message would be more correct, what do you think?

according to  https://www.matheboard.de/archive/592607/thread.html
ggt(0,0) is defined as 0

**Comment by t-m-z** [comment]
searchstrings corrected,
widget refactored
regarding 0 - these are defined by number theory

**Pull request merged**